pkg/policy/api/rule_validation_test.go

```diff
@@ -686,6 +686,102 @@ func (s *PolicyAPITestSuite) TestTooManyPortsRule(c *C) {
 	c.Assert(err, NotNil)
 }
 
+func (s *PolicyAPITestSuite) TestTooManyICMPFields(c *C) {
+	var fields []ICMPField
+
+	for i := 1; i <= 1+maxICMPFields; i++ {
+		fields = append(fields, ICMPField{
+			Type: uint8(i),
+		})
+	}
+
+	tooManyICMPRule := Rule{
+		EndpointSelector: WildcardEndpointSelector,
+		Ingress: []IngressRule{
+			{
+				IngressCommonRule: IngressCommonRule{
+					FromEndpoints: []EndpointSelector{WildcardEndpointSelector},
+				},
+				ICMPs: ICMPRules{{
+					Fields: fields,
+				}},
+			},
+		},
+	}
+	err := tooManyICMPRule.Sanitize()
+	c.Assert(err, NotNil)
+}
+
+func (s *PolicyAPITestSuite) TestWrongICMPFieldFamily(c *C) {
+	wrongFamilyICMPRule := Rule{
+		EndpointSelector: WildcardEndpointSelector,
+		Ingress: []IngressRule{
+			{
+				IngressCommonRule: IngressCommonRule{
+					FromEndpoints: []EndpointSelector{WildcardEndpointSelector},
+				},
+				ICMPs: ICMPRules{{
+					Fields: []ICMPField{{
+						Family: "hoge",
+						Type:   0,
+					}},
+				}},
+			},
+		},
+	}
+	err := wrongFamilyICMPRule.Sanitize()
+	c.Assert(err, NotNil)
+}
+
+func (s *PolicyAPITestSuite) TestICMPRuleWithOtherRuleFailed(c *C) {
+	ingressICMPWithPort := Rule{
+		EndpointSelector: WildcardEndpointSelector,
+		Ingress: []IngressRule{
+			{
+				IngressCommonRule: IngressCommonRule{
+					FromEndpoints: []EndpointSelector{WildcardEndpointSelector},
+				},
+				ToPorts: []PortRule{{
+					Ports: []PortProtocol{
+						{Port: "80", Protocol: ProtoTCP},
+					},
+				}},
+				ICMPs: ICMPRules{{
+					Fields: []ICMPField{{
+						Type: 8,
+					}},
+				}},
+			},
+		},
+	}
+
+	egressICMPWithPort := Rule{
+		EndpointSelector: WildcardEndpointSelector,
+		Egress: []EgressRule{
+			{
+				EgressCommonRule: EgressCommonRule{
+					ToEndpoints: []EndpointSelector{WildcardEndpointSelector},
+				},
+				ToPorts: []PortRule{{
+					Ports: []PortProtocol{
+						{Port: "80", Protocol: ProtoTCP},
+					},
+				}},
+				ICMPs: ICMPRules{{
+					Fields: []ICMPField{{
+						Type: 8,
+					}},
+				}},
+			},
+		},
+	}
+
+	err := ingressICMPWithPort.Sanitize()
+	c.Assert(err, NotNil)
```

Comment:
I think we can use the `ErrorMatches` matcher here to assert on the actual error returned